contracts/standard/arbitration/ArbitrableDeposit.sol

```diff
@@ -7,74 +6,178 @@
+    uint public claimDepositAmount; // Total amount a claimant must deposit. 
+
+    enum Status {NoDispute, WaitingOwner, WaitingClaimant, DisputeCreated, Resolved}
+    Status public status;
+    
+    uint8 constant AMOUNT_OF_CHOICES = 2;
+    uint8 constant OWNER_WINS = 1;
+    uint8 constant CLAIMANT_WINS = 2;
+    string constant RULING_OPTIONS = "Owner wins;Claimant wins"; // A plain English of what rulings do. Need to be redefined by the child class.
+
+    modifier onlyOwner{ require(msg.sender==address(owner)); _; }
+    modifier onlyNotOwner{ require(msg.sender!=address(owner)); _;}
+    modifier onlyClaimant{ require(msg.sender==address(claimant)); _;}
 
-    uint public claimAmount; // Amount claimed by claimant
-    uint public claimRate; // Rate of a claim the claimant must deposit as an integer
-    uint internal claimResponseAmount; // Amount the claimant is granted by the owner
+    enum Party {Owner, Claimant}
 
-    function ArbitrableDeposit (
-        Arbitrator _arbitrator, bytes32 _hashContract, uint _timeout,
-        address _partyB, bytes _arbitratorExtraData)
-        TwoPartyArbitrable(_arbitrator,_hashContract,_timeout,_partyB,_arbitratorExtraData) public payable {
+    /** @dev Indicate that a party has to pay a fee or would otherwise be considered as loosing.
+     *  @param _party The party who has to pay.
+     */
+    event HasToPayFee(Party _party);
+
+    /** @dev Constructor. Choose the arbitrator
+     *  @param _arbitrator The arbitrator of the contract.
+     *  @param _hashContract Keccak256 hash of the plain text contract.
+     *  @param _timeout Time after which a party automatically loose a dispute.
+     *  @param _arbitratorExtraData Extra data for the arbitrator.
+     */
+    function ArbitrableDeposit (Arbitrator _arbitrator, bytes32 _hashContract, uint _timeout, bytes _arbitratorExtraData, uint _claimRate) Arbitrable(_arbitrator, _arbitratorExtraData, _hashContract) public payable {
+        timeout = _timeout;
+        claimRate = _claimRate;
+        status = Status.NoDispute;
         amount += msg.value;
+        owner = msg.sender;
+        address(this).transfer(amount);
     }
 
-    /** @dev Owner deposit to contract. To be called when the owner
+    /** @dev Owner deposit to contract. To be called when the owner makes a deposit.
      */
-    function deposit() onlyPartyA {
-        address(this).transfer(amount);
+    function deposit(uint _amount) onlyOwner {
+        amount += _amount;
+        address(this).transfer(_amount);
     }
     
-    function claim(uint _claimValue) onlyPartyB {
-        require(_claimValue >= 0 && _claimValue <= amount);
-        claimAmount = _claimValue;
-        address(this).transfer((_claimValue * claimRate)/100);
+    /** @dev File a claim against owner. To be called when someone makes a claim.
+     *  @param _claimAmount The proposed claim amount by the claimant.
+     */    
+    function makeClaim(uint _claimAmount) onlyNotOwner {
```

Comment:
This need to be payable to have a claimDeposit.
We must also ask the claimer to deposit arbitration fee upfront to avoid frivolous claims.